### ISSUE TITLE
Spring instanciation broken for MatchesListRegexDecideRule

### ISSUE THREAD

**Issue by anjackson** [internetarchive/heritrix3#337]
In [this context](https://github.com/internetarchive/heritrix3/pull/290#discussion_r366711640) a lambda function was introduced to [MatchesListRegexDecideRule](https://github.com/internetarchive/heritrix3/blame/ef38dc08c357bcb2353bbfbcd00a2c3370aad9c7/modules/src/main/java/org/archive/modules/deciderules/MatchesListRegexDecideRule.java#L117).

This class no longer works when created as part of the crawler beans -- instead producing [this inscrutable error](https://stackoverflow.com/a/46963786/6689).

As far as I can tell, this should not work unless we upgrade Spring, so I'm rather confused as to how this came about.

**Comment by ato** [comment]
Yikes! Might be good motivation for merging #255. I assume an anonymous inner class would be a  workaround?

```java
FutureTask<Boolean> matchesFuture = new FutureTask<>(new Callable<Boolean>() {
    // We can't use a lambda as our ancient version of Spring can't cope (see #337)
    public Boolean call() {
        return p.matcher(interruptible).matches();
    }
});
```

> As far as I can tell, this should not work unless we upgrade Spring, so I'm rather confused as to how this came about.

Curious. I could understand if it was just through the code review changes but the lambda was in the original PR too. morokosi also patched it in #328 presumably without hitting this error.

Hmm. By any chance did you hit this on a top-level named bean? Perhaps Spring only does it's class parsing shenanigans for named beans. I notice the instance in the default Heritrix config doesn't have an id but I see a few in UKWA's configs that do.

CC @csrster

**Comment by anjackson** [comment]
@ato yes, this was on a named bean. I'll test your _anonymous inner class_ version when I get chance.
